src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowValidationNegativeSpec.groovy

```diff
@@ -149,47 +163,43 @@ class FlowValidationNegativeSpec extends HealthCheckSpecification {
         when: "Create a flow with protected path"
         def switchPair = switchPairs.all().neighbouring().withAtLeastNNonOverlappingPaths(2).random()
-        def flow = flowHelperV2.randomFlow(switchPair)
-        flow.allocateProtectedPath = true
-        flowHelperV2.addFlow(flow)
+        def flow = flowFactory.getBuilder(switchPair)
+                .withProtectedPath(true)
+                .build().create()
 
         then: "Flow with protected path is created"
-        northbound.getFlowPath(flow.flowId).protectedPath
+        flow.retrieveAllEntityPaths().getPathNodes(Direction.FORWARD, true)
 
         and: "Validation of flow with protected path must be successful"
-        northbound.validateFlow(flow.flowId).each { direction ->
+        flow.validate().each { direction ->
             assert direction.discrepancies.empty
         }
 
         when: "Delete rule of protected path on the srcSwitch"
-        def flowPathInfo = northbound.getFlowPath(flow.flowId)
-        def protectedPath = flowPathInfo.protectedPath.forwardPath
-        def rules = northbound.getSwitchRules(switchPair.src.dpId).flowEntries.findAll {
-            !new Cookie(it.cookie).serviceFlag
-        }
-
+        def flowPathInfo = flow.retrieveAllEntityPaths()
+        def protectedPath = flowPathInfo.getPathNodes(Direction.FORWARD, true)
+        def swRules = switchRulesFactory.get(switchPair.src.dpId)
+        def rules = swRules.getRules().findAll { !new Cookie(it.cookie).serviceFlag }
         def ruleToDelete = rules.find {
-            it.instructions?.applyActions?.flowOutput == protectedPath[0].inputPort.toString() &&
-                    it.match.inPort == protectedPath[0].outputPort.toString()
+            it.match.inPort == protectedPath[0].portNo.toString()
         }.cookie
-
-        switchHelper.deleteSwitchRules(switchPair.src.dpId, ruleToDelete)
+        swRules.delete(ruleToDelete)
 
         then: "Flow validate detects discrepancies"
         //TODO(andriidovhan) try to extend this test when the issues/2302 is fixed
-        def responseValidateFlow = northbound.validateFlow(flow.flowId).findAll { !it.discrepancies.empty }*.discrepancies
+        def responseValidateFlow = flow.validate().findAll { !it.discrepancies.empty }*.discrepancies
         assert responseValidateFlow.size() == 1
         responseValidateFlow[0].expectedValue[0].contains(ruleToDelete.toString())
 
         when: "Delete all rules except default on the all involved switches"
-        def mainPath = flowPathInfo.forwardPath
-        def involvedSwitchIds = (mainPath*.switchId + protectedPath*.switchId).unique()
+        def flowPath = flow.retrieveAllEntityPaths()
+        def involvedSwitchIds = flowPath.getInvolvedSwitches()
         involvedSwitchIds.each { switchId ->
-            switchHelper.deleteSwitchRules(switchId, DeleteRulesAction.IGNORE_DEFAULTS)
+            switchRulesFactory.get(switchId).deleteAllCookies(DeleteRulesAction.IGNORE_DEFAULTS)
```

Comment:
It seems like we cannot change  `switchHelper.deleteSwitchRules()` call to  `switchRulesFactory.get(switchId).deleteAllCookies()` as we omit SYNC action for these switches in the scope of overall cleanup.